enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+
+### Future work
+
+There are a number of topics in this area which would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and its concrete behaviour, for example add a "capability" for machine-api
+2. Precisely define reaction of the operators listed below for the "External" platform type 
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+
+### Affected components
+
+This section enumerates OpenShift's components and briefly elaborates the future plans around this proposal.
+During initial implementation we must ensure that all OpenShift's components treat "External" platform in the same way as
+"None" to ensure we do not disrupt the current openshift operation.
+
+In the future we will need to change the behavior of OpenShift components to be able to receive supplemental provider-specific components from an infrastructure provider
+or, if a component manages something else (i.e. kubelet, kcm), adjust its behaviour (set `--cloud-provider=external` arg to kubelet for example).
+
+Specific component changes will be described in details within separate enhancement documents on a per-component basis.
+
+#### Machine Config Operator
+
+Currently, MCO sets kubelet flags based on a [PlatformStatus and PlatformType](https://github.com/openshift/machine-config-operator/blob/7babc03f84088f49072c6f5b0d1769850d09a694/pkg/controller/template/render.go#L343).
+Such flag is [crucial](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) for Cloud Controller Manager engagement within the cluster.
+
+Initially, the new "External" platform should be treated similarly to PlatformType "None" by the MCO, and do not set up any cloud-specific flags for the kubelet.
+Then, down the road (during [phase 3](#implementation-phases)), it would be expected for the MCO to use the "External" platform type and its spec as a signal about the underlying platform and cloud controller manager presence and operate accordingly.
+For explicit signal about the necessity to set `--cloud-provider=external` flag to the kubelet, we could use the `CloudControllerManager` of the `ExternalPlatformSpec`, which is 
+described in [API Extensions](#api-extensions) section down below.
+
+#### Kube Controller Manager Operator
+
+Same as the [MCO](#machine-config-operator) in regard to kubelet, Kube Controller Manager Operator manages KCM (kube-controller-manager) deployments.
+
+Historically Kube Controller Manager was a home for cloud-specific control loops. This logic is also engaging by setting up proper
+flags on KCM executable like,
+```shell
+...
+--cloud-provider=azure
+--cloud-config=/etc/kubernetes/cloud.conf
+...
+```
+
+For engaging an external Cloud Controller Manager no `cloud-provider` flag should be set for the KCM executable.
```

Comment:
So? Does KCM actually need to know anything here? Sounds like KCMO just needs to know it's external and a CCM is installed, and set nothing, sounds quite easy? But then, if a CCM isn't installed, it has the same behaviour? So, does KCM actually care about this or can it behave like platform None always?

Reply:
>But then, if a CCM isn't installed, it has the same behaviour? So, does KCM actually care about this or can it behave like platform None always?

It doesn't care and can always behave as for None platform.

Actually kube doc says that `--cloud-provider` flag must not be set to the kcm, but as i can judge from this code: https://github.com/openshift/library-go/blob/f946ce665f858b213b687f40480ac34d0b26987e/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L79-L87 flag sets there.

Reply:
Ok cool, so make sure to include that the KCMO doesn't need to do anything when platform is external, wrt to the cloud provider flag. I had thought it needed to check the infra object but looks like no